client/src/components/App.js

```diff
@@ -7,25 +7,28 @@ import "../assets/scss/main.scss";
 import RegistrationForm from "./registration/RegistrationForm";
 import SignInForm from "./authentication/SignInForm";
 import TopBar from "./layout/TopBar";
+import CategoryIndex from './CategoryIndex';
 
 const App = (props) => {
-  const [currentUser, setCurrentUser] = useState(undefined);
+  const [currentUser, setCurrentUser] = useState(null);
```

Comment:
We do want this initial state to be `undefined`, it is how we can differentiate between pre-fetch and post-fetch state